catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/data/impl/MetacardTypeFactory.java

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.data.impl;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import ddf.catalog.data.AttributeDescriptor;
+import ddf.catalog.data.MetacardType;
+import ddf.catalog.data.impl.taxonomy.CoreAttributes;
+import ddf.catalog.data.taxonomy.MetacardComponent;
+
+public class MetacardTypeFactory {
+
+    /**
+     * Generates a {@link MetacardType} that contains all of the attributes of
+     * provided {@link MetacardComponent}s
+     *
+     * @param name     the name of the {@link MetacardType}
+     * @param typeList the list of {@link MetacardComponent}s
+     * @return
+     */
+    public static MetacardType createMetacardType(String name, List<MetacardComponent> typeList) {
```

Comment:
I think this could just be a constructor on `MetacardTypeImpl`.  We could also do a factory method on `MetacardType` using default methods.